File: Leetcode/sbgo/Leetcode1218-最长定差序列.go
package sbgo

func longestSubsequence(arr []int, difference int) int {
	f := map[int]int{}
	ans := 1
	for _, a := range arr {
		t, err := f[a-difference]
		if err {
			f[a] = t + 1
		} else {
			f[a] = 1
		}
		ans = max(ans, f[a])
	}
	return ans
}

func max(i, j int) int {
	if i < j {
		return j
	}
	return i
}

File: Leetcode/PY/Leetcode1218-最长定差序列.py
class Solution:
    def longestSubsequence(self, arr: List[int], difference: int) -> int:
        f = defaultdict(int)
        ans = 1
        for a in arr:
            f[a] = f[a - difference] + 1
            ans = max(ans,f[a])
        return ans